fairing/preprocessors/converted_notebook.py

```diff
@@ -55,3 +74,61 @@ def preprocess(self):
             f.write(contents)
         self.executable = converted_notebook
         return [converted_notebook]
+
+
+class ConvertNotebookPreprocessorWithFire(ConvertNotebookPreprocessor):
+    """Create an entrpoint using pyfire."""
+    def __init__(self,
+                 class_name=None,
+                 notebook_file=None,
+                 notebook_preprocessor=FilterIncludeCell,
+                 executable=None,
+                 command=["python"],
+                 path_prefix=constants.DEFAULT_DEST_PREFIX,
+                 output_map=None):
+
+        super().__init__(
+            notebook_file=notebook_file,
+            notebook_preprocessor=notebook_preprocessor,
+            executable=executable,
+            command=command,
+            path_prefix=path_prefix,
+            output_map=output_map)
+
+        self.class_name = class_name
+
+    def preprocess(self):
+        exporter = nbconvert.PythonExporter()
+        exporter.register_preprocessor(self.notebook_preprocessor, enabled=True)
+        processed, _ = exporter.from_filename(self.notebook_file)
+
+        lines = []
+        for l in processed.splitlines():
+            # Get rid of multiple blank lines
+            if not l.strip():
+                if lines:
+                    if not lines[-1]:
+                        # last line is already blank don't add another one
+                        continue
+            # strip in statements
+            if l.startswith("# In["):
+                continue
+            lines.append(l)
+
+        contents = "\n".join(lines)
+        converted_notebook = Path(self.notebook_file).with_suffix('.py')
```

Comment:
can you do this with a tempfile or check for existing file with the same name? writing a .py in the same dir might overwrite something the user already has.

Reply:
Okay, I also updated the original preprocessor.